src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -3,54 +3,23 @@
 import static core.basesyntax.Main.actionHandlerMap;
 
 import core.basesyntax.Main;
-import core.basesyntax.model.Account;
+import core.basesyntax.model.FruitTransaction;
+
+import java.util.List;
 
 public class ShopServiceImpl implements ShopService {
-    private ActionStrategy strategy = new ActionStrategyImpl(actionHandlerMap);
+    private ActionStrategy strategy = new ActionStrategyImpl();
 
     @Override
-    public void generate(String[] infoFromDatabase) {
-        for (int i = 0; i < infoFromDatabase.length; i++) {
-            generateByRow(infoFromDatabase[i]);
+    public void generate(List<FruitTransaction> listOfAllTransactions) {
+        for (int i = 0; i < listOfAllTransactions.size(); i++) {
+            generateByRow(listOfAllTransactions.get(i));
         }
     }
 
-    void generateByRow(String row) {
+    void generateByRow(FruitTransaction transaction) {
```

Comment:
this method is broken I guess, it doesn't do any proper logic

Reply:
Agree

Reply:
please reply, it's a void method without side effects, what does it do?